ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Block/SupportsPeras.hs

```diff
@@ -79,13 +149,29 @@ class
 
   data PerasCert blk
 
+  data PerasVote blk
+
   data PerasValidationErr blk
 
+  data PerasForgeErr blk
+
   validatePerasCert ::
     PerasCfg blk ->
     PerasCert blk ->
     Either (PerasValidationErr blk) (ValidatedPerasCert blk)
 
+  validatePerasVote ::
+    PerasCfg blk ->
+    PerasVote blk ->
+    PerasVoteStakeDistr ->
+    Either (PerasValidationErr blk) (ValidatedPerasVote blk)
+
+  forgePerasCert ::
+    PerasCfg blk ->
+    PerasVoteTarget blk ->
+    [ValidatedPerasVote blk] ->
+    Either (PerasForgeErr blk) (PerasCert blk)
```

Comment:
```suggestion
    Either (PerasForgeErr blk) (ValidatedPerasCert blk)
```

I stand by my point made [here](https://github.com/IntersectMBO/ouroboros-consensus/pull/1768#discussion_r2591745079) that we probably want to consider a certificate forged by ourselves that didn't return any error as a Validated cert :)